src/Serilog.Sinks.AzureDocumentDb/Sinks/AzureDocumentDb/AzureDocumentDbSink.cs

```diff
@@ -35,42 +36,37 @@ public AzureDocumentDBSink(Uri endpointUri, string authorizationKey, string data
             _formatProvider = formatProvider;
             _client = new DocumentClient(endpointUri, authorizationKey);
             _storeTimestampInUtc = storeTimestampInUtc;
-            Task.WaitAll(new []{CreateDatabaseIfNotExistsAsync(databaseName)});
-            Task.WaitAll(new []{CreateCollectionIfNotExistsAsync(collectionName)});
+            CreateDatabaseIfNotExistsAsync(databaseName).ConfigureAwait(false);
```

Comment:
If the .ConfigureAwait(false) calls are made within the Create* methods (where the await keyword appears) we don't need the calls up here (Though .Wait() is still necessary).